modules/Hoot/ui/modals/importDataset.js

```diff
@@ -414,61 +418,78 @@ export default class ImportDataset {
             folderId
         };
 
-        this.processRequest = Hoot.api.uploadDataset( data )
-            .then( resp => {
-                this.loadingState();
-
-                this.jobId = resp.data[ 0 ].jobid;
-
-                return Hoot.api.statusInterval( this.jobId );
-            } )
-            .then( async resp => {
-                let message;
-                if (resp.data && resp.data.status === 'cancelled') {
-                    message = 'Import job cancelled';
-
-                    // Delete the newly created folder
-                    if ( newFolderName ) {
-                        await Hoot.api.deleteFolder( folderId );
-                        await Hoot.folders.removeFolder( folderId );
-                    }
-
-                    this.submitButton
-                        .select( 'span' )
-                        .text( 'Import' );
-                } else {
-                    message = 'Import job complete';
+        this.loadingState();
+
+        this.processRequest =  Hoot.api.uploadDataset( data, this.source )
+        .then( resp => {
+            this.jobId = resp.data[ 0 ].jobid;
+            return Hoot.api.statusInterval( this.jobId );
+        } )
+        .then( async resp => {
+            let message;
+
+            if ( resp.message === 'Request canceled'  ) {
+                Hoot.message.alert( {
+                    message: resp.message,
+                    type: 'warn'
+                } );
+            }
+
+            if (resp.data && resp.data.status === 'cancelled') {
+                message = 'Import job cancelled';
+
+                // Delete the newly created folder
+                if ( newFolderName ) {
+                    await Hoot.api.deleteFolder( folderId );
+                    await Hoot.folders.removeFolder( folderId );
                 }
 
+                this.submitButton
+                    .select( 'span' )
+                    .text( 'Import' );
+            } else {
+                message = 'Import job complete';
+            }
+
+            Hoot.message.alert( {
+                data: resp.data,
+                message: message,
+                status: 200,
+                type: resp.type
+            } );
+
+            return resp;
+        } )
+        .then( () => Hoot.folders.refreshAll() )
+        .then( () => Hoot.events.emit( 'render-dataset-table' ) )
+        .catch( err => {
+            console.error(err);
+
+            if ( err.message === 'Request cancelled.' ) {
                 Hoot.message.alert( {
-                    data: resp.data,
-                    message: message,
-                    status: 200,
-                    type: resp.type
+                    message: 'Upload request cancelled',
```

Comment:
might as well use `err.message` here and make it match what we want to show the user.